### PR TITLE
[feature](Nereids) add password func

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes
add password function
```
select password("123");
+-------------------------------------------+
| password('123')                           |
+-------------------------------------------+
| *23AE809DDACAF96AF0FD78ED04B6A265E05AA257 |
+-------------------------------------------+
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.59 seconds
 stream load tsv:          546 seconds loaded 74807831229 Bytes, about 130 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          67 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17161933076 Bytes

File: regression-test/suites/nereids_function_p0/scalar_function/P.groovy [hunk 1]
```diff
@@ -27,6 +27,7 @@ suite("nereids_scalar_fn_P") {
 	sql "select parse_url(kvchrs1, 'HOST', 'PROTOCOL') from fn_test_not_nullable order by kvchrs1, kvchrs1, kvchrs1"
 	sql "select parse_url(kstr, 'HOST', 'PROTOCOL') from fn_test order by kstr, kstr, kstr"
 	sql "select parse_url(kstr, 'HOST', 'PROTOCOL') from fn_test_not_nullable order by kstr, kstr, kstr"
+	qt_sql_pwd """select password("123")"""
```
Comment (morrySnow) [inline regression-test/suites/nereids_function_p0/scalar_function/P.groovy:30]: add qt_sql_password_Varchar, qt_sql_password_String, qt_sql_password_Varchar_notnull, qt_sql_password_String_notnull

**Comment by XieJiann** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.51 seconds
 stream load tsv:          547 seconds loaded 74807831229 Bytes, about 130 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          30.2 seconds inserted 10000000 Rows, about 331K ops/s
 storage size: 17161988215 Bytes

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Review by sohardforaname — APPROVED** [review]

**Pull request merged**